### PR TITLE
--log option and new logger level

### LINKED ISSUE #184: Add new level to logger
For now we have 4 levels: `[INFO]` `[DEBUG]` `[WARN]` `[ERROR]`, however now it's a lot of debug information, which could be useful only in rare cases, but generally it just loads output. E.g.: execution output of `ProcessBuilder`

It's suggested to create new additional level `[TRACE]` for such things

### PR DESCRIPTION
**Pull request by sanyavertolet**
## log option and new logger level

This pull request closes #184 

## Actions checklist
* [ ] Added --log option - now you choose a level of logging.
* [ ] Editted README.md
* [ ] Removed --quiet option.
* [ ] Removed --debug option.
* [ ] fixed a problem with native part - each time we print anything to stdout/stderr, we create a new instance of heavy structure - FILE, and never destroyed any of them.
* [ ] improved logging when SAVE cannot find any input file that matches a pattern

### CODE REVIEW

**Review by akuleshov7 — APPROVED** [review]

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt [hunk 1]
```diff
@@ -144,7 +145,7 @@ data class TestConfig(
                 .filter { plugin ->
                     plugin.discoverTestFiles(directory).any().also { isNotEmpty ->
                         if (!isNotEmpty) {
-                            logDebug("Plugin <${plugin::class.simpleName}> in config file ${plugin.testConfig.location} has no test resources; " +
+                            logTrace("Plugin <${plugin::class.simpleName}> in config file ${plugin.testConfig.location} has no test resources; " +
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt]: Ok, done

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/ExtraFlagsExtractor.kt [hunk 2]
```diff
@@ -40,7 +40,7 @@ class ExtraFlagsExtractor(private val generalConfig: GeneralConfig,
         .let(ExtraFlags::from)
         .also {
             if (it == ExtraFlags("", "")) {
-                logWarn("Line <$line> is matched by extraFlagsPattern <${generalConfig.runConfigPattern}>, but no flags have been extracted")
+                logTrace("Line <$line> is matched by extraFlagsPattern <${generalConfig.runConfigPattern}>, but no flags have been extracted")
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/ExtraFlagsExtractor.kt]: Makes sence. Done.

File: examples/kotlin-diktat/fix_and_warn/smoke/src/main/kotlin/org/cqfn/save/chapter2/Bug2Expected.kt [hunk 3]
```diff
@@ -26,7 +25,6 @@ internal fun createWithFile(
     containerName: String,
     file: File,
     resources: Collection<File> = emptySet()
-    // ;warn:30:3: [EMPTY_BLOCK_STRUCTURE_ERROR] incorrect format of empty block: empty blocks are forbidden unless it is function with override keyword (cannot be auto-corrected)
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply examples/kotlin-diktat/fix_and_warn/smoke/src/main/kotlin/org/cqfn/save/chapter2/Bug2Expected.kt]: Done

File: examples/kotlin-diktat/fix_and_warn/smoke/src/main/kotlin/org/cqfn/save/chapter1/Bug1Expected.kt [hunk 4]
```diff
@@ -1,19 +1,12 @@
-// ;warn:1:1: [FILE_NAME_MATCH_CLASS] file name is incorrect - it should match with the class described in it if there is the only one class declared: Bug1Expected.kt vs D
 package org.cqfn.save.chapter1
 
-// ;warn:4:1: [MISSING_KDOC_TOP_LEVEL] all public and internal top-level classes and functions should have Kdoc: D (cannot be auto-corrected)
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply examples/kotlin-diktat/fix_and_warn/smoke/src/main/kotlin/org/cqfn/save/chapter1/Bug1Expected.kt]: Done

File: save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt [hunk 5]
```diff
@@ -106,13 +108,33 @@ class Save(
                    |$excludedNote
                 """.trimMargin()
             } else {
-                "SAVE wasn't able to run tests, please check the correctness of configuration and test resources"
+                for (item in testConfigs) {
+                    logWarn(item::class.simpleName ?: "")
+                }
+                val fixPluginPatterns = testConfigs
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt]: Ok, done

File: save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt [hunk 6]
```diff
@@ -106,13 +108,33 @@ class Save(
...
+                    .last()
+                    .pluginConfigs
+                    .filterIsInstance<FixPluginConfig>()
+                    .map { it.resourceNamePatternStr }
+                    .distinct()
+                    .joinToString(", ")
+                    .ifEmpty { "NONE" }
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt]: Yes we do, removed it

File: save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt [hunk 7]
```diff
@@ -106,13 +108,33 @@ class Save(
                    |$excludedNote
                 """.trimMargin()
             } else {
-                "SAVE wasn't able to run tests, please check the correctness of configuration and test resources"
+                for (item in testConfigs) {
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt]: Done

**Pull request merged**